icu4j/perf-tests/src/main/java/com/ibm/icu/dev/test/perf/PerfTest.java

```diff
@@ -937,7 +938,9 @@ public BOMFreeReader(InputStream in, String encoding) throws IOException {
             if (amountRead > bomLength)
                 pushback.unread(start, bomLength, amountRead - bomLength);
 
-            reader = (encoding == null) ? new InputStreamReader(pushback) : new InputStreamReader(pushback, encoding);
+            reader = (encoding == null)
+                    ? new InputStreamReader(pushback, StandardCharsets.UTF_8)
```

Comment:
It looks like the existing code uses the default/system charset intentionally when no explicit one is given.

Reply:
Done

What we discussed. Throw for null or invalid encoding.